### PR TITLE
show search dropdown menu with "How to search" link

### LINKED ISSUE #1513: Make it easier to find the search query syntax documentation
As a user who is performing searches, I want to be able to quickly understand what kind of queries Sourcegraph supports. It takes ~3-4 clicks to get to the query syntax doc page (https://sourcegraph.com/help/user/search/queries#diff-and-commit-searches-only).

### PR DESCRIPTION
**Pull request by sqs**
fix #1513 

In the future we can add more to help the user construct their query (recent searches, saved searches, quick reference, etc.).

screenshot from 2018-12-23 14-58-28
screenshot from 2018-12-23 14-58-23
screenshot from 2018-12-23 14-58-05

### CODE REVIEW

File: web/src/search/input/SearchButton.scss [hunk 1]
```diff
@@ -1,13 +1,21 @@
 .search-button {
-    margin-left: 0.5rem;
-
     &__label {
         margin-left: 0.25rem;
     }
 
+    .btn {
+        // Override Bootstrap's fake focus ring, which looks weird.
```
Comment (felixfbecker) [inline web/src/search/input/SearchButton.scss]: override with what?

**Comment by francisschmaltz** [comment]
I do like this mechanic though and I think it's worth exploring in the future. Might be a good way to quickly enable or disable search extensions.

**Comment by sqs** [comment]
The PR description has more future things I wanted to put in there (pretty soon). I’ll revise this PR to mock that up and repost for feedback. I’ll also see about making the down-caret a ? instead.

**Comment by francisschmaltz** [comment]
@sqs there should be both a dropdown and a question mark, that way the two are separate actions

**Comment by sqs** [comment]
Check out the updated screenshots. The help icon is back. It opens a dropdown. I plan to put a quick reference in the dropdown, but I posted it before that's done to give a preview.

File: web/src/search/input/SearchButton.scss (hunk 1 above)
Comment (sqs) [reply web/src/search/input/SearchButton.scss]: Removed this override.

**Comment by felixfbecker** [comment]
I still have the same complaint that it is part of the button, and the primary action (executing the search) is no longer the right-most action. It breaks with the input-with-search-button-on-the-side UI pattern that's been everywhere on the internet pages for ages. It also makes me feel like it's harder to hit the search button with the mouse, even if that is not actually true if it keeps the same size, the presence of a different action button glued right next to it with no margin in between subconsciously makes me be more careful when clicking it to not accidentally hit the other button (vs only risking hitting empty space).

**Comment by sqs** [comment]
OK, here it is. If you don't like it, please suggest another way to fix #1513.

screenshot from 2018-12-23 16-37-33

**Comment by sqs** [comment]
Going to merge soon

**Comment by sqs** [comment]
Slight update, will merge soon:

screenshot from 2018-12-31 00-13-28

**Comment by francisschmaltz** [comment]
The icon should be smaller - inline -with the bottom link

**Comment by francisschmaltz** [comment]
@sqs We should roll this back and take a look at the UI for it. It breaks /start explore, and the UI for the search bar itself is off. 

screen shot 2019-01-03 at 2 39 14 pm

**Pull request merged**